back/boxtribute_server/graph_ql/resolvers.py

```diff
@@ -219,26 +219,33 @@ def resolve_qr_code(obj, _, qr_code=None):
 
 
 @box.field("tags")
-def resolve_box_tags(box_obj, _):
-    return (
-        Tag.select()
-        .join(TagsRelation)
-        .where(
-            (TagsRelation.object_id == box_obj.id)
-            & (TagsRelation.object_type == TaggableObjectType.Box)
-        )
-    )
+def resolve_box_tags(box_obj, info):
+    return info.context["tags_for_box_loader"].load(box_obj.id)
 
 
 @query.field("product")
+def resolve_product(obj, _, id):
+    product = Product.get_by_id(id)
```

Comment:
why aren't data loaders used here as well? 
Couldn't be there a query which at first queries the product e.g. as a subfield of a box related query, but later on it's also called on top-level (root query level)?

Reply:
Your're right, dataloaders can be used in many more places (and should be). My focus rn was to improve the resolvers for the slow queries.

Honestly also here I was having trouble to load a product and then pass it to the subsequent authorize call, probably have to await it :grimacing: